### PR TITLE
Prevent "Unsupported tarball" exception upon download of Git dependency containing problematic symlinks

### PR DESCRIPTION
**Pull request by dynamicboid**
When Stack projects contain `extra-dep` Git dependencies that have certain problematic symlinks, `stack` will throw an error when building a project if it is unable to resolve those symlinks successfully. However, on subsequent invocations of Stack, it seems that Pantry is able to cache the Git artifacts and enable Stack to run successfully. This PR seeks to address the former issue of Stack failing upon the first invocation on such a project. This may thus partially address issues [#4913](https://github.com/commercialhaskell/stack/issues/4913), [#5004](https://github.com/commercialhaskell/stack/issues/5004) and [#5603](https://github.com/commercialhaskell/stack/issues/5603) on the Stack repository as well as [#23](https://github.com/monadfix/shower/issues/23) on the Haskell `shower` repo.

### CODE REVIEW

File: src/Pantry/Archive.hs [hunk 1]
```diff
@@ -356,6 +356,10 @@ data SimpleEntry = SimpleEntry
   }
   deriving Show
 
+removeInitialDotSlash :: FilePath -> FilePath
+removeInitialDotSlash filename = 
+  maybe filename T.unpack (T.stripPrefix "./" $ T.pack filename)
```
Comment (neilmayhew) [inline src/Pantry/Archive.hs:361]: There's a `stripPrefix` function in `Data.List`, so I don't think you need to convert to and from `Text` here.

Comment (mpilgrem) [reply src/Pantry/Archive.hs:361]: @neilmayhew, I put this through.

**Comment by mpilgrem** [comment]
Merged manually with d8132749a5d8bfe807f04c15ebf8c8a4827be1a6.

Rather than normalise `getFiles`, it seemed to me better to add normalisation to the `ATZip` limb of `foldArchive`, as `foldTar` already normalises the other two limbs.

**Comment by neilmayhew** [comment]
cc @philderbeast

File: src/Pantry/Archive.hs (hunk 1 above)
Comment (neuroevolutus) [reply src/Pantry/Archive.hs:361]: A much belated thanks, Neil.

**Comment by neuroevolutus** [comment]
@mpilgrem, thanks so much for taking the time to review this!

**Pull request closed without merging**